packages/components/src/internal/components/domainproperties/actions.ts

```diff
@@ -905,7 +905,7 @@ export function updateDataType(field: DomainField, value: any): DomainField {
             field = field.merge({
                 lookupValidator: PropDescType.isUser(value)
                     ? new PropertyValidator(LOOKUP_VALIDATOR_VALUES)
-                    : undefined,
+                    : field.lookupValidator,
```

Comment:
nit: Could also consider non-mutating check:

```ts
} else if (PropDescType.isUser(value)) {
    field = field.merge({ lookupValidator: new PropertyValidator(LOOKUP_VALIDATOR_VALUES) }) as DomainField;
}
```

Reply:
Ah yes. That's better.